File: backend/src/main/java/org/gestionstock/stock/Payload/Response/QuoteResponse.java
package org.gestionstock.stock.Payload.Response;

import java.util.List;

public record QuoteResponse(
    String id,
    String quoteName,
    String description,
    CompanyResponse company,
    ContactResponse contact,
    List<ProductQuoteResponse> productQuote,
    String createdAt,
    String updatedAt,
    String createdBy,
    String updatedBy
) {
    
}

File: frontend/src/app/models/response/quote-response.ts
import { CompanyResponse } from "./company-response";
import { ContactResponse } from "./contact-response";
import { ProductQuoteResponse } from "./product-quote-response";

export interface QuoteResponse {
    id: string;
    quoteName: string;
    description: string;
    company: CompanyResponse;
    contact: ContactResponse;
    productQuote: ProductQuoteResponse[];
    createdAt: Date;
    updatedAt: Date;
    createdBy: string;
    updatedBy: string;
}
